src/tools/pbs_python.c

```diff
@@ -2310,11 +2310,37 @@ main(int argc, char *argv[], char *envp[])
 		char *largv[3];
 		int ll;
 		char *pc, *pc2;
+
+#ifdef SYSTEM_PYTHON_PATH
+		snprintf(python_path, MAXPATHLEN, "%s", SYSTEM_PYTHON_PATH);
+		pc = strdup(SYSTEM_PYTHON_PATH);
+		if(pc == NULL) {
+			fprintf(stderr, "Out of memory\n");
+			return 1;
+		}
+		pc2 = strstr(pc,"bin/python");
+		if(pc2 == NULL) {
+			fprintf(stderr, "Python executable not found!\n");
+			return 1;
+		}
+		ll = pc2 - pc;
+		if(ll <= 0) {
+			fprintf(stderr, "Python home not found!\n");
+			return 1;
+		}
+		snprintf(python_prefix, ll, "%s", SYSTEM_PYTHON_PATH);
```

Comment:
This is slightly confusing that you're using snprintf() to only copy part of the string (instead of using the second argument as to not overrun a buffer).  The usual way the server handles something like this is to set a '\0' byte in your string and then use it.